apps/src/music/views/MusicView.jsx

```diff
@@ -527,6 +537,20 @@ class UnconnectedMusicView extends React.Component {
     return this.props.levelProperties?.exemplarSources;
   };
 
+  getExemplarPlaybackEvents = () => {
```

Comment:
Might be good to calculate this once per level and store in an instance variable, so we're not creating a new workspace and sequencer every time? We can have a `this.exemplarPlaybackEvents` here in `MusicView`, and in `onLevelLoad`, we can call this function to set that field.

Then, instead of passing a getter to the ExemplarPlayerView, we can just pass the events directly since they will be updated whenever the level changes. And for the `MusicValidator`, the getter can just be a function that returns `this.exemplarPlaybackEvents`, which will still be updated whenever the level loads.

Reply:
Updated!